resolver/serviceResolver.go

```diff
@@ -232,8 +232,8 @@ func (resolver *serviceResolver) VerifyCode(userAddress sdkCore.AddressHandler,
 }
 
 // SignMessage validates user's message, then adds guardian signature and returns the message.
-func (resolver *serviceResolver) SignMessage(userAddress sdkCore.AddressHandler, userIp string, request requests.SignMessage) ([]byte, *requests.OTPCodeVerifyData, error) {
-	guardian, otpCodeVerifyData, err := resolver.verifyCodesReturningGuardian(userAddress, request.GuardianAddr,
+func (resolver *serviceResolver) SignMessage(userIp string, request requests.SignMessage) ([]byte, *requests.OTPCodeVerifyData, error) {
+	guardian, otpCodeVerifyData, err := resolver.verifyCodesReturningGuardian(request.UserAddr, request.GuardianAddr,
```

Comment:
here you can create the address like this:

```go
userAddress, err := sdkData.NewAddressFromBech32String(request.UserAddr)
if err != nil {
	return nil, nil, err
}
```

this way, you don't need to further decode the address in `verifyCodesReturningGuardian`, which can keep the same form as before, as it is used from `validateTxRequestReturningGuardian`

Reply:
done